### ISSUE TITLE
[ci.jenkins.io] Container agents in a degraded state

### ISSUE THREAD

**Issue by basil** [jenkins-infra/helpdesk#2893]
### Service(s)

ci.jenkins.io

### Summary

I've now run 3 core PR builds in a row against a Linux JDK 8 Kubernetes container (label: `maven`) and all three have failed with:

```
Cannot contact jnlp-maven-8-45549: hudson.remoting.ChannelClosedException: Channel "hudson.remoting.Channel@3ba7fb1d:JNLP4-connect connection from ec2-3-143-84-83.us-east-2.compute.amazonaws.com/3.143.84.83:64651": Remote call on JNLP4-connect connection from ec2-3-143-84-83.us-east-2.compute.amazonaws.com/3.143.84.83:64651 failed. The channel is closing down or has closed down`
```

- https://ci.jenkins.io/job/Core/job/jenkins/job/PR-6499/2/console
- https://ci.jenkins.io/job/Core/job/jenkins/job/PR-6499/3/console
- https://ci.jenkins.io/job/Core/job/jenkins/job/PR-6499/4/console

https://status.jenkins.io/ shows no issues.

Clearly the cluster is in a severely degraded state and monitoring is insufficient.

### Reproduction steps

Try to run a core PR build. Observe that the build fails within a minute rather than compiling and testing the code.

**Comment by basil** [comment]
The acute problem I was facing seems to have now cleared up:

- https://ci.jenkins.io/job/Core/job/jenkins/job/PR-6499/5/console

This ticket can be closed, since the short-term pain is gone. I am still trying to think of the best way to file a ticket to track the long-term pain of agent instability we are seeing.

**Comment by dduportal** [comment]
Some elements to add:

- There is a lot of jnlp agent that are in "idle" state. It's weird, no idea why/what/how.
- I have zero idea how to monitor such behaviors. Maybe collecting all logs and have a sytem grepping it for such errors? Monitoring Jenkins have always been a pain, any idea welcome to be honest.
- DockerHub suffered issues ~2 hours ago, it was timeout to pull images: might be related.

**Comment by dduportal** [comment]
> https://status.jenkins.io/ shows no issues.

It's basically a static website: there isn't really automation on this. Do not hesitate to open a PR on it if you see such pain in agents allocation: that would help other users a lot.

**Comment by dduportal** [comment]
For info, I've triggered a "reload from disk". It usually trigger whatever "agent garbage collector" process in Jenkins usually.

**Comment by dduportal** [comment]
Currently checking the logs to see what is going wrong

**Comment by dduportal** [comment]
A lot of errors 
```
io.jenkins.plugins.casc.ConfiguratorException: Invalid configuration elements for type class org.csanchez.jenkins.plugins.kubernetes.ContainerTemplate : env.
Available attributes : alwaysPullImage, args, command, envVars, image, livenessProbe, name, ports, privileged, resourceLimitCpu, resourceLimitEphemeralStorage, resourceLimitMemory, resourceRequestCpu, resourceRequestEphemeralStorage, resourceRequestMemory, runAsGroup, runAsUser, shell, ttyEnabled, workingDir
```

**Comment by dduportal** [comment]
Stopped puppet on the VM to validate the correct casc syntax.

[async ping @lemeurherve @smerle33 : it looks like that your work on doks is also generating a lot of errors. It will be deleted by the operation I'm doing right now]

**Comment by basil** [comment]
> […] Invalid configuration elements […] : env. Available attributes : […] envVars […]

=> https://github.com/jenkins-infra/jenkins-infra/pull/2129

**Comment by dduportal** [comment]
Many thanks @basil for reporting the issue + sending the PR! I've added a suggestion that I just tested in the production.

**Comment by dduportal** [comment]
- Puppet enabled
- @basil PR merged ❤️ 
- Puppet currently applying change, monitoring the logs.

**Comment by dduportal** [comment]
- Change applied with success. New pods are triggered (casc reload went well). 
- However, still a lot of "suspended" pod. Weird state, with a LOT of error messages like
```
903de-aeb0-410d-ad22-8c8eb88071a5', name='jnlp-maven-11', slaveConnectTimeout=100, label='container kubernetes cik8s maven-11 jdk11', containers=[ContainerTemplate{name='jnlp', image='jenkinsciinfra/inbound-agent-maven@sha256:85af372d080c2e4d15d55f8833a6eb5a84f3f206368da002a048ccf49e302c23', workingDir='/home/jenkins', command='/usr/local/bin/jenkins-agent', args='', resourceRequestCpu='4', resourceRequestMemory='8G', resourceRequestEphemeralStorage='', resourceLimitCpu='4', resourceLimitMemory='8G', resourceLimitEphemeralStorage='', livenessProbe=ContainerLivenessProbe{execArgs='', timeoutSeconds=0, initialDelaySeconds=0, failureThreshold=0, periodSeconds=0, successThreshold=0}}], imagePullSecrets=[PodImagePullSecret{name='dockerhub-credential'}]}
2022-04-20 20:04:23.768+0000 [id=718347]	INFO	o.c.j.p.k.KubernetesSlave#_terminate: Terminating Kubernetes instance for agent jnlp-maven-11-kh5b0
2022-04-20 20:04:23.796+0000 [id=718347]	SEVERE	o.c.j.p.k.KubernetesSlave#_terminate: Computer for agent is null: jnlp-maven-11-kh5b0
2022-04-20 20:04:23.796+0000 [id=718347]	INFO	hudson.slaves.AbstractCloudSlave#terminate: FATAL: Computer for agent is null: jnlp-maven-11-kh5b0
2022-04-20 20:04:28.017+0000 [id=734210]	INFO	h.TcpSlaveAgentListener$ConnectionHandler#run: Accepted JNLP4-connect connection #28,715 from /3.143.84.83:51530
```

But no traces of these  pod in the UI or in the kubernetes pod list.

**Comment by dduportal** [comment]
VM is unresponsive (all CPUs at 100%, Jenkins is waiting on I/Os). Triggering a reboot.

**Comment by dduportal** [comment]
Applied last puppet changes / plugins upgrades / Jenkins container restart

**Comment by dduportal** [comment]
Aaaaaand we are DockerHub rate limited of course

**Comment by dduportal** [comment]
All pods deleted. Jenkins created a set of new ones \o/

But still getting rate limit from DockerHub. Gotta wait for the next quota in a ~2 hours.

**Comment by basil** [comment]
Sounds like a good time for me to triage UX regressions. =)

**Comment by dduportal** [comment]
ci.jenkins.io is restarted again, machine was 100% CPU, with a thousands builds in the queue. Still the BOm builds :'(

**Comment by dduportal** [comment]
OK, it seems that ci.jenkins.io keeps restarting the BOM PR builds.
The following error message happens a lot in the logs:
```
Queue item for node block in Tools » bom » PR-1034 #1 is missing (perhaps JENKINS-34281); rescheduling
```

**Comment by dduportal** [comment]
After restarting the VM, had to delete the `$JENKINS_HOME/nodes/jnlp-*` to have it starting without error.
Seems fine now.

**Comment by dduportal** [comment]
As per #2894, there were other issues (job scannings not being triggered). Another proper restart of the container in the VM seems to solve this issue.

Let's wait for https://github.com/jenkins-infra/helpdesk/issues/2866 to be applied to both Kubernetes cluster before triggering a bom build again.

**Comment by dduportal** [comment]
Currently operating the data disk of the VM:
- Resizing from 300 to 500Gb as it was filled at 91% (could do a garbage collection but Jenkins being stopped, it is hard). The goal is to get this partition below the dreaded 80% threshold (performances)
- Changing disk type from "HDD" to "premium SSD" to boost performances

**Comment by dduportal** [comment]
Jenkins is starting: currently scanning the whole JENKINS_HOME, gotta take some times

**Comment by basil** [comment]
("Infinite increase" in consumption is a common pattern, unfortunately encouraged by cloud providers who stand to make a large amount of profit from this. Maintaining existing levels consumption or decreasing consumption is much harder. I think this is because increases in consumption are an effect typically several layers removed from the cause. Tracing an increase in consumption back to the cause (especially in a distributed system) is significantly more difficult than e.g. debugging a Java stack trace, where each element in the causal chain is presented in the error message. That additional difficulty, in my opinion, explains why people often choose to increase their consumption instead.)

**Comment by dduportal** [comment]
- ci.jenkins.io is back
- both EKS and Digital ocean clusters are re-configured (and upgraded to Kube 1.21) and ready
but stil:

- Kubernetes agent pods are created, running but Jenkins keeps not using them. No error obvious error logs. I have no idea what to do now.

**Comment by basil** [comment]
Agents seem to be slowly coming up in https://ci.jenkins.io/job/Tools/job/bom/job/PR-1032/2/consoleFull so ramp-up appears to be in progress. Assuming we don't eventually overload the system and crash, so far so good. (If we do, having captured thread dumps during the ramp-up period would help in debugging.) Let me know if you want to debug together over a screen share.

**Comment by dduportal** [comment]
Many thanks for the help proposal @basil . It looks like that the ramp up is in progress, let's wait ~30 min to see how it behaves, now that the clusters are at full scale.

I wonder if we shouldn't add a lock on the BOM builds, cross-branches, given the huge impact it has each time there is a build storm. WDYT?

**Comment by basil** [comment]
Possibly, depending on the cause. I do not yet fully understand the symptoms of the problem here, let alone the cause. It is possible that the robustness or scalability of the software (i.e., OSS Jenkins core and plugins) could be improved, which would be a more difficult task but would have much higher long-term value than implementing deployment-specific workarounds in the form of locks.

**Comment by dduportal** [comment]
The queue is (slowly) decreasing:

<img width="1694" alt="Capture d’écran 2022-04-21 à 19 04 21" src="https://user-images.githubusercontent.com/1522731/164513428-f1c2d0a9-262a-4e2d-9099-04f1dab28ebe.png">

**Comment by basil** [comment]
But only because jobs like https://ci.jenkins.io/job/Tools/job/bom/job/PR-1037/2/console are hitting https://github.com/jenkinsci/workflow-durable-task-step-plugin/blob/master/src/main/java/org/jenkinsci/plugins/workflow/support/pickles/ExecutorPickle.java#L131-L133= and being aborted. That in and of itself is a issue: the job should have resumed without interruption, but it did not (something I have complained about before). This issue has the perverse side effect of driving down load and possibly hiding other issues.

**Comment by jetersen** [comment]
I think we broke it again 😓

**Comment by dduportal** [comment]
> But only because jobs like https://ci.jenkins.io/job/Tools/job/bom/job/PR-1037/2/console are hitting https://github.com/jenkinsci/workflow-durable-task-step-plugin/blob/master/src/main/java/org/jenkinsci/plugins/workflow/support/pickles/ExecutorPickle.java#L131-L133= and being aborted. That in and of itself is a issue: the job should have resumed without interruption, but it did not (something I have complained about before). This issue has the perverse side effect of driving down load and possibly hiding other issues.

I'm sorry @basil but I have no idea what is this "pickle" stuff about. I trust you that you identify what does it causes but it make no sense to me 😅

**Comment by dduportal** [comment]
> I think we broke it again 😓

@jetersen Do you have a job not starting or waiting since hours?

**Comment by jetersen** [comment]
@dduportal 
https://github.com/jenkinsci/bom/pull/1029 https://ci.jenkins.io/blue/organizations/jenkins/Tools%2Fbom/detail/PR-1029/2/pipeline
https://github.com/jenkinsci/bom/pull/1035 https://ci.jenkins.io/blue/organizations/jenkins/Tools%2Fbom/detail/PR-1035/1/pipeline
https://github.com/jenkinsci/bom/pull/1038 https://ci.jenkins.io/blue/organizations/jenkins/Tools%2Fbom/detail/PR-1038/1/pipeline

**Comment by dduportal** [comment]
> @dduportal [jenkinsci/bom#1029](https://github.com/jenkinsci/bom/pull/1029) https://ci.jenkins.io/blue/organizations/jenkins/Tools%2Fbom/detail/PR-1029/2/pipeline [jenkinsci/bom#1035](https://github.com/jenkinsci/bom/pull/1035) https://ci.jenkins.io/blue/organizations/jenkins/Tools%2Fbom/detail/PR-1035/1/pipeline [jenkinsci/bom#1038](https://github.com/jenkinsci/bom/pull/1038) https://ci.jenkins.io/blue/organizations/jenkins/Tools%2Fbom/detail/PR-1038/1/pipeline

Currently checking and running these builds one at a time (I understand there are long term fixes but each wave of dependanbot PRs on the BOM is taking a toll on ci.jenkins.io but having a lock to only allow 1 PR build of the BOM at a time for now would help a lot, given the amount of power that this builds requires).

**Comment by dduportal** [comment]
OK, so even with 1 bom build, still this "weird" behavior of suspended pods, while the pods are running (with their logs telling us it is connected to Jenkins controller). These pods are trashed after a 5 min timeout, with no build handled.

No obvious log so not sure what is happening and how to troublshoot this.

**Comment by basil** [comment]
> I'm sorry @basil but I have no idea what is this "pickle" stuff about. I trust you that you identify what does it causes but it make no sense to me :sweat_smile:

Identifying the cause of failures in an application requires being able to read and understand the application code. If you are uncomfortable doing that, no problem; simply decline to do a retrospective. I am happy to take on the task if desired.

> We need to […] add a lock to sequentialize these builds.

As stated in https://github.com/jenkins-infra/helpdesk/issues/2893#issuecomment-1105461838, it is premature to come to a conclusion about what action to take without clearly defining the symptoms of the problem and its cause.

**Comment by basil** [comment]
Yes abstraction is required, but with a firm grasp of the fundamentals one can zoom in and out from high-level to low-level as needed, learning the new abstractions just-in-time as one adjusts the zoom level. In my experience there is very little room for shade when it comes to analytical reasoning and root cause analysis. Absolute logic has served me well in that domain.

I actually started to get involved in the Jenkins community when finding the cause of a similar operational failure back in 2018. Amusingly, you can read my early (confused) emails about it here:

- https://groups.google.com/g/jenkinsci-dev/c/NG1Jn08KYEM/m/mac1eEarBgAJ
- https://groups.google.com/g/jenkinsci-dev/c/zrTkBTWtvhI/m/jLraS0XRBAAJ

Eventually I was able to find the cause and submit a detailed bug report and test case, and Jesse Glick fixed the bug a year later in [JENKINS-41854](https://issues.jenkins.io/browse/JENKINS-41854). There was much rejoicing at my company when that issue was resolved.

In any case I agree that we need to take some short to medium term action. I will think about what to do over the next week.

**Comment by dduportal** [comment]
You are totally right (in my own opinion) and the concern is improving the overall quality of Jenkins. If we hit that issue, for sure other users are.

I only want to add multiple ideas, with different time scopes, to the whole thing.

Let's work on a retrospective next week after dust have settled and everyone's mind was garbage collected by the week-end :)

Many thanks for the insights, the help and work!

**Comment by lemeurherve** [comment]
Noticed this PR, seems related to our problem: [(Jenkins-68126) Remove watcher to fix Jenkins agents in suspended state after upgrade to 2.332.1 with kubernetes agents, queued builds not executing](https://github.com/jenkinsci/kubernetes-plugin/pull/1167)

**Comment by dduportal** [comment]
For info: post-mortem meeting by the team proposed (should happen wednesday the 03rd of May if everyone is available, otherwise in the upcoming days)

**Comment by sbeaulie** [comment]
@lemeurherve have you been able to try out our PR, it has been working well for us in production. We have also made other startup performance improvements and will open a separate PR.

**Comment by sbeaulie** [comment]
This also really helped our infra run things quickly https://github.com/jenkinsci/kubernetes-plugin/pull/1171

**Comment by dduportal** [comment]
Thanks for the feedback @sbeaulie ! Alas we haven't been able to try this in production. The main reason is that the root cause of the problem is now gone for us (Docker API rate limit, leading into a lot of retries).

**Comment by dduportal** [comment]
Draft of the post-mortem (collaborative notes that would be moved in GitHub once "frozen"): https://hackmd.io/8IGwo3QXSa-kcuPmKLDTQg.

Last call @basil @lemeurherve @smerle33 @MarkEWaite for review and approval: ETA Monday 30th of May.

**Comment by dduportal** [comment]
Pleas finde below the notes from the team post-mortem (was due since a few weeks):

# Post-Mortem of Outages on ci.jenkins.io (April/May 2022)

* Outage on April 2?, 2022
* Outage on May 2, 2022


## Attendees

* Mark Waite
* Basil Crow
* Damien DUPORTAL

## Introduction

Thanks to @basil, a preliminary analysis work was done prior to this meeting, providing an initial guidance.
The sections of these notes materialize the sections of this preliminary as it was the direction of the meeting.

Initial helpdesk issues:

- Outage 1: https://github.com/jenkins-infra/helpdesk/issues/2893
- Outage 2: https://github.com/jenkins-infra/helpdesk/issues/2908

## General Process

* File separate ticket to track long-term resolution of each issue (problem -> evaluation -> solution -> implementation).
* If a short-term workaround is needed, file a separate ticket to cover the reason a workaround is needed, the implementation of the workaround, and a third ticket to cover removal of the workaround.
  * Question from Basil: "do you agree on this process?"
      * Both Mark and Damien agree on keeping it: it solves the "keep information"
  * Let's use the GitHub issue feature that permit creating issues from comment's bullet list (hover your cursor on the rendered bullet list and click the button that appears on the right of the bullet list item). 
    It provides a back link to original issue: useful to keep knowledge

## Agent disconnections


Observable with an error message like the following:

```console
Cannot contact jnlp-maven-8-45549: hudson.remoting.ChannelClosedException: Channel "hudson.remoting.Channel@3ba7fb1d:JNLP4-connect connection from ec2-3-143-84-83.us-east-2.compute.amazonaws.com/3.143.84.83:64651": Remote call on JNLP4-connect connection from ec2-3-143-84-83.us-east-2.compute.amazonaws.com/3.143.84.83:64651 failed. The channel is closing down or has closed down`
```

@basil have seen this problem in many core and plugin PRs. This is fatal to builds, requiring builds to be restarted. This drives up costs.

At a high level, the debugging process goes like this: when this problem is occurring, examine both the controller and the agent:

* If the build is still running on the agent, then determine why the connection was interrupted and why the agent did not reconnect to the controller after the connection was interrupted. (The connection could have been interrupted due to a networking issue or due to resource exhaustion on the agent or controller. The connection may have failed to resume due to a software bug.)
* If the build is not still running on the agent, determine why it stopped. (The build may have stopped running on the agent due to resource exhaustion on the agent.)


There seems to be 3 different patterns of disconnections:

* EC2/Azure VM agents
* Windows containers (on ACS)
* Linux containers on (EKS/DoKS)


**Action Items** 

* (short-term)TODO: give access to Kubernetes clusters of ci.jenkins.io (EKS and DOKS) to Basil
  * Helpdesk Issue for tracking: <https://github.com/jenkins-infra/helpdesk/issues/2946>
* (short-term): Catch this problem in the act (not hard, just requires patience) and start debugging as described above.
* (medium-term): TBD based on analysis
* (long-term): TBD based on analysis

## Deployment of untested change on the infra

Issue [jenkins-infra/jenkins-infra#2128](https://github.com/jenkins-infra/jenkins-infra#2128)

Caused errors like this:

```console
io.jenkins.plugins.casc.ConfiguratorException: Invalid configuration elements for type class org.csanchez.jenkins.plugins.kubernetes.ContainerTemplate : env.
Available attributes : alwaysPullImage, args, command, envVars, image, livenessProbe, name, ports, privileged, resourceLimitCpu, resourceLimitEphemeralStorage, resourceLimitMemory, resourceRequestCpu, resourceRequestEphemeralStorage, resourceRequestMemory, runAsGroup, runAsUser, shell, ttyEnabled, workingDir
```

The cause is clear: the change was not tested and contains a syntax error. Fixed by collaborting efficiently, thanks y'all!

**Action Items** 

* (short-term): None, the syntax was corrected.
* (medium-term): We should do proper testing in the future. :-)
  * Team-wide discussion:
    * somes changes are costly or hard to test (JCasc yaml can be linted, but we need trigger an environment to test a given element), so the strategy is to iterate quickly, on tiny amounts so rollbacks are non-events and easy. However breaking things is frustrating for users so room for improvement team-wide.
    * There isn't a staging environment (cost and maintenance reasons)
    * For Kubernetes-related changes we absolutely should improve as it's easy to spin up k3s clusters on PRs
    * For puppet, there is a [vagrant based testing](https://github.com/jenkins-infra/jenkins-infra#vagrant-based-testing) that involves spawning a virtualbox VM with vagrant, where the puppet profiles can be applied (but only on Intel machines) + rspec unit tests
* (long-term): Improve contribution process to make testing easier.


## Low disk space

We resized the disk from 300 to 500 GiB as it was filled at 91%.
The goal was to get this partition below the dreaded 80% threshold for performance.

**Action Items** 

* (short-term): None, the disk has been resized.
* (medium-term): Collect the output of sudo ncdu -o /tmp/output.txt -x ${JENKINS_HOME} and begin analysis. Already done by @basil
* (long-term): audit of the jenkins_home disk usage:
  * Some branches of jobs are producing quite the amount of data
  * There is a UX Core PR producing a LOT of logs.
      * Install "log file size" plugin (https://plugins.jenkins.io/logfilesizechecker/) if needed.


## Suspended Kubernetes Agents

Many "suspended" nodes with errors like the below but no traces of these pods in the controller UI. In the Kubernetes pod list, all pods are in a valid state with no log errors.

```console
903de-aeb0-410d-ad22-8c8eb88071a5', name='jnlp-maven-11', slaveConnectTimeout=100, label='container kubernetes cik8s maven-11 jdk11', containers=[ContainerTemplate{name='jnlp', image='jenkinsciinfra/inbound-agent-maven@sha256:85af372d080c2e4d15d55f8833a6eb5a84f3f206368da002a048ccf49e302c23', workingDir='/home/jenkins', command='/usr/local/bin/jenkins-agent', args='', resourceRequestCpu='4', resourceRequestMemory='8G', resourceRequestEphemeralStorage='', resourceLimitCpu='4', resourceLimitMemory='8G', resourceLimitEphemeralStorage='', livenessProbe=ContainerLivenessProbe{execArgs='', timeoutSeconds=0, initialDelaySeconds=0, failureThreshold=0, periodSeconds=0, successThreshold=0}}], imagePullSecrets=[PodImagePullSecret{name='dockerhub-credential'}]}
2022-04-20 20:04:23.768+0000 [id=718347]    INFO    o.c.j.p.k.KubernetesSlave#_terminate: Terminating Kubernetes instance for agent jnlp-maven-11-kh5b0
2022-04-20 20:04:23.796+0000 [id=718347]    SEVERE    o.c.j.p.k.KubernetesSlave#_terminate: Computer for agent is null: jnlp-maven-11-kh5b0
2022-04-20 20:04:23.796+0000 [id=718347]    INFO    hudson.slaves.AbstractCloudSlave#terminate: FATAL: Computer for agent is null: jnlp-maven-11-kh5b0
2022-04-20 20:04:28.017+0000 [id=734210]    INFO    h.TcpSlaveAgentListener$ConnectionHandler#run: Accepted JNLP4-connect connection #28,715 from /3.143.84.83:51530
```

**Action Items** 

* Possible cause: https://issues.jenkins.io/browse/JENKINS-68126 (thanks Hervé!)
* => We feel that DockerHub API rate limit increase should push away this issue. But if it happens again, could be worth it to diagnose deeper.

* (short-term): Try to reproduce in a staging environment; if reproducible, verify that the analysis from JENKINS-68126 matches the thread dump from the Jenkins controller.
* (medium-term): If analysis matches, proceed with development and deployment of fix for JENKINS-68126; otherwise, analyze further by connecting a debugger to Jenkins controller
* (long-term): Based on analysis

=> Priority: let's park this issue in background as it does not seem to be blocking and prone to happen

## Jenkins unresponsive

VM is unresponsive (all CPUs at 100%, Jenkins is waiting on I/Os). Triggered a reboot.

**Action Items** 

* (short-term): None, service has been restored.
* (medium-term): Develop a runbook for how to collect debugging state when CPU exhaustion occurs. At a high level, get a thread dump and CPU flame graph. The latter requires the image to be preconfigured with async-profiler.
* (long-term): TBD based on analysis


## Proposal from Infra team

* If we can pinpoint the issue to the "dependabot" peak on sundays, then maybe add a lock on BOM
  * To be challenged though: we need facts for this. By design Jenkins should be able to handle this but might be bugs to be fixed in certain cases

* Build of other contributors competing with BOM builds if we test a peak?
  * Can we add prioritization through a plugin or specific setup (Priority Sorter Plugin)?
  * Add VM templates for labels but makes it harder to diagnose problems
  * Why not splitting workloads on different clusters to ensure build are treated?


Overall: Infra team is waiting for @basil's feedbacks or requests (access to systems, installations, etc.), unless of course not enough time.


[Post Meeting]
- https://github.com/jenkins-infra/helpdesk/issues/2912 raised by Basil, implemented recently, related to adding rngd (to improve random number generation efficiency). Done by the infra team.

**Comment by dduportal** [comment]
Closing as no more outages + no more actions left for now.

Tracking improvement on the kubernetes plugin in https://github.com/jenkins-infra/helpdesk/issues/2964
